posix/fdpass.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * File descriptor passing
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Ziemowit Leszczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "fdpass.h"
+
+
+/* FIXME: _Alignof is available since C11 */
+#define CMSG_ALIGN(n)       (((n) + _Alignof(struct cmsghdr) - 1) & ~(_Alignof(struct cmsghdr) - 1))
```

Comment:
I think it would be safe to assume standard alignment of this struct - 8 bytes. I really don't like introducing C11 syntax (when we're, especially in kernel, are trying to adhere to C89)

Reply:
I've switched to GCC __alignof__